joblib/_multiprocessing_helpers.py

```diff
@@ -22,18 +24,34 @@
 #            issue a warning if not
 if mp is not None:
     try:
-        # Use the spawn context
-        if sys.version_info < (3, 3):
-            Semaphore = mp.Semaphore
+        if sys.platform == "win32":
+            # Use the spawn context
+            if PY27:
+                Semaphore = mp.Semaphore
+            else:
+                # Using mp.Semaphore has a border effect and set the default
+                # backend for multiprocessing. To avoid that, we use the
+                # 'spawn' context which is available on all supported platforms
+                ctx = mp.get_context('spawn')
+                Semaphore = ctx.Semaphore
+            _sem = Semaphore()
```

Comment:
we need to `del _sem` right away here, no?

Reply:
The `del` operation was further away to be common to both `win32` and other platform but with the unique implementation, I used it right away.